integration/tests/access/cohort1/access_api_test.go

```diff
@@ -191,6 +200,116 @@ func (s *AccessAPISuite) TestMVPScriptExecutionLocalStorage() {
+	conn, err := grpc.Dial(accessNodeContainer.Addr(testnet.GRPCPort), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	s.Require().NoError(err)
+	s.Require().NotNil(conn)
+
+	// Create a client for the access API
+	accessClient := accessproto.NewAccessAPIClient(conn)
+	serviceClient, err := accessNodeContainer.TestnetClient()
+	s.Require().NoError(err)
+	s.Require().NotNil(serviceClient)
+
+	// Get the latest block ID
+	latestBlockID, err := serviceClient.GetLatestBlockID(s.ctx)
+	s.Require().NoError(err)
+
+	// Generate a new account transaction
+	accountKey := test.AccountKeyGenerator().New()
+	payer := serviceClient.SDKServiceAddress()
+
+	tx, err := templates.CreateAccount([]*sdk.AccountKey{accountKey}, nil, payer)
+	s.Require().NoError(err)
+	tx.SetComputeLimit(1000).
+		SetReferenceBlockID(sdk.HexToID(latestBlockID.String())).
+		SetProposalKey(payer, 0, serviceClient.GetSeqNumber()).
+		SetPayer(payer)
+
+	tx, err = serviceClient.SignTransaction(tx)
+	s.Require().NoError(err)
+
+	// Convert the transaction to a message format expected by the access API
+	authorizers := make([][]byte, len(tx.Authorizers))
+	for i, auth := range tx.Authorizers {
+		authorizers[i] = auth.Bytes()
+	}
+
+	convertToMessageSig := func(sigs []sdk.TransactionSignature) []*entities.Transaction_Signature {
+		msgSigs := make([]*entities.Transaction_Signature, len(sigs))
+		for i, sig := range sigs {
+			msgSigs[i] = &entities.Transaction_Signature{
+				Address:   sig.Address.Bytes(),
+				KeyId:     uint32(sig.KeyIndex),
+				Signature: sig.Signature,
+			}
+		}
+
+		return msgSigs
+	}
+
+	transactionMsg := &entities.Transaction{
+		Script:           tx.Script,
+		Arguments:        tx.Arguments,
+		ReferenceBlockId: tx.ReferenceBlockID.Bytes(),
+		GasLimit:         tx.GasLimit,
+		ProposalKey: &entities.Transaction_ProposalKey{
+			Address:        tx.ProposalKey.Address.Bytes(),
+			KeyId:          uint32(tx.ProposalKey.KeyIndex),
+			SequenceNumber: tx.ProposalKey.SequenceNumber,
+		},
+		Payer:              tx.Payer.Bytes(),
+		Authorizers:        authorizers,
+		PayloadSignatures:  convertToMessageSig(tx.PayloadSignatures),
+		EnvelopeSignatures: convertToMessageSig(tx.EnvelopeSignatures),
+	}
+
+	// Send and subscribe to the transaction status using the access API
+	subClient, err := accessClient.SendAndSubscribeTransactionStatuses(s.ctx, &accessproto.SendAndSubscribeTransactionStatusesRequest{
+		Transaction: transactionMsg,
+	})
+	s.Require().NoError(err)
+
+	expectedCounter := uint64(0)
+	var finalTxStatus entities.TransactionStatus
+	var txID sdk.Identifier
+
+	for {
+		resp, err := subClient.Recv()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+
+			s.Require().NoError(err)
+		}
+
+		if txID == sdk.EmptyID {
+			txID = sdk.Identifier(resp.GetId())
+		}
+
+		s.Assert().Equal(expectedCounter, resp.GetMessageIndex())
+		s.Assert().Equal(txID, sdk.Identifier(resp.GetId()))
+
+		expectedCounter++
+		finalTxStatus = resp.Status
```

Comment:
can you have it also check that it receives notifications for each of the statuses in the processes